### PR TITLE
Feature/73964 project details draft

### PR DESCRIPTION
**Pull request by jakub-lotko**

### CODE REVIEW

File: src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/CreateProjectCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public CreateProjectCommandHandler(IApplicationProjectsRepository applicationProjectsRepository, ILoanUserContext loanUserContext)
+    {
+        _applicationProjectsRepository = applicationProjectsRepository;
+        _loanUserContext = loanUserContext;
+    }
+
+    public async Task<OperationResult<ProjectId>> Handle(CreateProjectCommand request, CancellationToken cancellationToken)
+    {
+        var applicationProjects = await _applicationProjectsRepository.GetById(request.Id, await _loanUserContext.GetSelectedAccount(), cancellationToken)
+            ?? throw new NotFoundException(nameof(ApplicationProjects), request.Id);
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/CreateProjectCommandHandler.cs:28]: Maybe this should be part of repository

File: src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ProjectCommandHandlerBase.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        var project = applicationProjects.Projects.FirstOrDefault(p => p.Id == projectId)
+            ?? throw new NotFoundException(nameof(Project), projectId);
+
+        try
+        {
+            action(project);
+        }
+        catch (DomainValidationException domainValidationException)
+        {
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ProjectCommandHandlerBase.cs:37]: Please log exception as warning. Check other base command handles

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Entities/ApplicationProjects.cs [hunk 3]
```diff
@@ -4,28 +4,33 @@
...
+
     public LoanApplicationId LoanApplicationId { get; }
 
-    public IList<Project> Projects { get; }
+    public IReadOnlyCollection<Project> Projects => _projects.AsReadOnly();
 
     public IList<Project> ActiveProjects => Projects.Where(p => !p.IsSoftDeleted).ToList();
 
-    public void AddProject()
+    public ProjectId AddProject()
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Entities/ApplicationProjects.cs:28]: AddProject -> AddEmptyProject

File: src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ChangeProjectNameCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+using System.Threading.Tasks;
+using HE.InvestmentLoans.BusinessLogic.LoanApplication.ApplicationProject.Repositories;
+using HE.InvestmentLoans.BusinessLogic.Projects.ValueObjects;
+using HE.InvestmentLoans.BusinessLogic.User;
+using HE.InvestmentLoans.Common.Extensions;
+using HE.InvestmentLoans.Common.Validation;
+using HE.InvestmentLoans.Contract.Projects.Commands;
+using MediatR;
+
+namespace HE.InvestmentLoans.BusinessLogic.Projects.CommandHandlers;
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ChangeProjectNameCommandHandler.cs:14]: I think it should be in "ApplicationProjects" namespace

File: src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/CreateProjectCommandHandler.cs (hunk 1 above)
Comment (jakub-lotko) [reply src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/CreateProjectCommandHandler.cs:28]: Yes, it should be

File: src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ChangeProjectNameCommandHandler.cs (hunk 4 above)
Comment (jakub-lotko) [reply src/HE.InvestmentLoans.BusinessLogic/Projects/CommandHandlers/ChangeProjectNameCommandHandler.cs:14]: ApplicationProjects namespace contains legacy code related to projects. All refactored code went to Projects namespace.

**Review by Carq — APPROVED** [review]

**Review by dawid-pawulski — APPROVED** [review]

**Pull request merged**